### PR TITLE
Change protocol confirmation function as this is now deprecated

### PR DESCRIPTION
**Pull request by ghost**
Change the protocol confirmation function because of a deprecated function.

`line 287: deprecated (protocol_confirmation): Remove in v5.1. Use analyzer_confirmation.`

### CODE REVIEW

File: analyzer/main.zeek [hunk 1]
```diff
@@ -284,7 +284,7 @@ function set_session(c: connection, message_id: int, opcode: LDAP::ProtocolOpcod
 }
 
 #############################################################################
-event protocol_confirmation(c: connection, atype: Analyzer::Tag, aid: count) &priority=5 {
+event analyzer_confirmation(c: connection, atype: Analyzer::Tag, aid: count) &priority=5 {
```
Comment (bbannier) [inline analyzer/main.zeek]: The rename from `protocol_confirmation` to `analyzer_confirmation` happened only with zeek-4.2.0, so let's keep the existing handler for compatibility with the Zeek LTS release.

```suggestion
@if (Version::at_least("4.2.0"))
event analyzer_confirmation(c: connection, atype: Analyzer::Tag, aid: count) {
@else
event protocol_confirmation(c: connection, atype: AllAnalyzers::Tag, aid: count) {
@endif
```

This requires `@load base/misc/version`.

**Review by bbannier — CHANGES_REQUESTED** [review]
Thanks for the PR. I left only one small issues for compatibility with zeek-4.0.

**Comment by ghost** [comment]
Good comment, I committed your suggestion.

**Review by bbannier — APPROVED** [review]
Thanks!

**Comment by bbannier** [comment]
Merged with 70b186e926a19501122fb7c616dc5008c195ac87.

**Pull request closed without merging**